File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/GasPhysicalQuantity.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class GasPhysicalQuantity : CommodityPhysicalQuantityBase
    {
        public GasPhysicalQuantity(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode physicalQuantityNode = xmlNode.SelectSingleNode("physicalQuantity");
            
            if (physicalQuantityNode != null)
            {
                if (physicalQuantityNode.Attributes["href"] != null || physicalQuantityNode.Attributes["id"] != null) 
                {
                    if (physicalQuantityNode.Attributes["id"] != null) 
                    {
                        physicalQuantityIDRef_ = physicalQuantityNode.Attributes["id"].Value;
                        CommodityNotionalQuantity ob = new CommodityNotionalQuantity(physicalQuantityNode);
                        IDManager.SetID(physicalQuantityIDRef_, ob);
                    }
                    else if (physicalQuantityNode.Attributes["href"] != null)
                    {
                        physicalQuantityIDRef_ = physicalQuantityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        physicalQuantity_ = new CommodityNotionalQuantity(physicalQuantityNode);
                    }
                }
                else
                {
                    physicalQuantity_ = new CommodityNotionalQuantity(physicalQuantityNode);
                }
            }
            
        
            XmlNode physicalQuantityScheduleNode = xmlNode.SelectSingleNode("physicalQuantitySchedule");
            
            if (physicalQuantityScheduleNode != null)
            {
                if (physicalQuantityScheduleNode.Attributes["href"] != null || physicalQuantityScheduleNode.Attributes["id"] != null) 
                {
                    if (physicalQuantityScheduleNode.Attributes["id"] != null) 
                    {
                        physicalQuantityScheduleIDRef_ = physicalQuantityScheduleNode.Attributes["id"].Value;
                        CommodityPhysicalQuantitySchedule ob = new CommodityPhysicalQuantitySchedule(physicalQuantityScheduleNode);
                        IDManager.SetID(physicalQuantityScheduleIDRef_, ob);
                    }
                    else if (physicalQuantityScheduleNode.Attributes["href"] != null)
                    {
                        physicalQuantityScheduleIDRef_ = physicalQuantityScheduleNode.Attributes["href"].Value;
                    }
                    else
                    {
                        physicalQuantitySchedule_ = new CommodityPhysicalQuantitySchedule(physicalQuantityScheduleNode);
                    }
                }
                else
                {
                    physicalQuantitySchedule_ = new CommodityPhysicalQuantitySchedule(physicalQuantityScheduleNode);
                }
            }
            
        
            XmlNode totalPhysicalQuantityNode = xmlNode.SelectSingleNode("totalPhysicalQuantity");
            
            if (totalPhysicalQuantityNode != null)
            {
                if (totalPhysicalQuantityNode.Attributes["href"] != null || totalPhysicalQuantityNode.Attributes["id"] != null) 
                {
                    if (totalPhysicalQuantityNode.Attributes["id"] != null) 
                    {
                        totalPhysicalQuantityIDRef_ = totalPhysicalQuantityNode.Attributes["id"].Value;
                        UnitQuantity ob = new UnitQuantity(totalPhysicalQuantityNode);
                        IDManager.SetID(totalPhysicalQuantityIDRef_, ob);
                    }
                    else if (totalPhysicalQuantityNode.Attributes["href"] != null)
                    {
                        totalPhysicalQuantityIDRef_ = totalPhysicalQuantityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        totalPhysicalQuantity_ = new UnitQuantity(totalPhysicalQuantityNode);
                    }
                }
                else
                {
                    totalPhysicalQuantity_ = new UnitQuantity(totalPhysicalQuantityNode);
                }
            }
            
        
            XmlNodeList minPhysicalQuantityNodeList = xmlNode.SelectNodes("minPhysicalQuantity");
            
            if (minPhysicalQuantityNodeList != null)
            {
                this.minPhysicalQuantity_ = new List<CommodityNotionalQuantity>();
                foreach (XmlNode item in minPhysicalQuantityNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            minPhysicalQuantityIDRef_ = item.Attributes["id"].Value;
                            minPhysicalQuantity_.Add(new CommodityNotionalQuantity(item));
                            IDManager.SetID(minPhysicalQuantityIDRef_, minPhysicalQuantity_[minPhysicalQuantity_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            minPhysicalQuantityIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        minPhysicalQuantity_.Add(new CommodityNotionalQuantity(item));
                        }
                    }
                    else
                    {
                        minPhysicalQuantity_.Add(new CommodityNotionalQuantity(item));
                    }
                }
            }
            
        
            XmlNodeList maxPhysicalQuantityNodeList = xmlNode.SelectNodes("maxPhysicalQuantity");
            
            if (maxPhysicalQuantityNodeList != null)
            {
                this.maxPhysicalQuantity_ = new List<CommodityNotionalQuantity>();
                foreach (XmlNode item in maxPhysicalQuantityNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            maxPhysicalQuantityIDRef_ = item.Attributes["id"].Value;
                            maxPhysicalQuantity_.Add(new CommodityNotionalQuantity(item));
                            IDManager.SetID(maxPhysicalQuantityIDRef_, maxPhysicalQuantity_[maxPhysicalQuantity_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            maxPhysicalQuantityIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        maxPhysicalQuantity_.Add(new CommodityNotionalQuantity(item));
                        }
                    }
                    else
                    {
                        maxPhysicalQuantity_.Add(new CommodityNotionalQuantity(item));
                    }
                }
            }
            
        
            XmlNode electingPartyNode = xmlNode.SelectSingleNode("electingParty");
            
            if (electingPartyNode != null)
            {
                if (electingPartyNode.Attributes["href"] != null || electingPartyNode.Attributes["id"] != null) 
                {
                    if (electingPartyNode.Attributes["id"] != null) 
                    {
                        electingPartyIDRef_ = electingPartyNode.Attributes["id"].Value;
                        PartyReference ob = new PartyReference(electingPartyNode);
                        IDManager.SetID(electingPartyIDRef_, ob);
                    }
                    else if (electingPartyNode.Attributes["href"] != null)
                    {
                        electingPartyIDRef_ = electingPartyNode.Attributes["href"].Value;
                    }
                    else
                    {
                        electingParty_ = new PartyReference(electingPartyNode);
                    }
                }
                else
                {
                    electingParty_ = new PartyReference(electingPartyNode);
                }
            }
            
        
        }
        
    
        #region PhysicalQuantity_
        private CommodityNotionalQuantity physicalQuantity_;
        public CommodityNotionalQuantity PhysicalQuantity_
        {
            get
            {
                if (this.physicalQuantity_ != null)
                {
                    return this.physicalQuantity_; 
                }
                else if (this.physicalQuantityIDRef_ != null)
                {
                    physicalQuantity_ = IDManager.getID(physicalQuantityIDRef_) as CommodityNotionalQuantity;
                    return this.physicalQuantity_; 
                }
                else
                {
                      return this.physicalQuantity_; 
                }
            }
            set
            {
                if (this.physicalQuantity_ != value)
                {
                    this.physicalQuantity_ = value;
                }
            }
        }
        #endregion
        
        public string physicalQuantityIDRef_ { get; set; }
        #region PhysicalQuantitySchedule_
        private CommodityPhysicalQuantitySchedule physicalQuantitySchedule_;
        public CommodityPhysicalQuantitySchedule PhysicalQuantitySchedule_
        {
            get
            {
                if (this.physicalQuantitySchedule_ != null)
                {
                    return this.physicalQuantitySchedule_; 
                }
                else if (this.physicalQuantityScheduleIDRef_ != null)
                {
                    physicalQuantitySchedule_ = IDManager.getID(physicalQuantityScheduleIDRef_) as CommodityPhysicalQuantitySchedule;
                    return this.physicalQuantitySchedule_; 
                }
                else
                {
                      return this.physicalQuantitySchedule_; 
                }
            }
            set
            {
                if (this.physicalQuantitySchedule_ != value)
                {
                    this.physicalQuantitySchedule_ = value;
                }
            }
        }
        #endregion
        
        public string physicalQuantityScheduleIDRef_ { get; set; }
        #region TotalPhysicalQuantity_
        private UnitQuantity totalPhysicalQuantity_;
        public UnitQuantity TotalPhysicalQuantity_
        {
            get
            {
                if (this.totalPhysicalQuantity_ != null)
                {
                    return this.totalPhysicalQuantity_; 
                }
                else if (this.totalPhysicalQuantityIDRef_ != null)
                {
                    totalPhysicalQuantity_ = IDManager.getID(totalPhysicalQuantityIDRef_) as UnitQuantity;
                    return this.totalPhysicalQuantity_; 
                }
                else
                {
                      return this.totalPhysicalQuantity_; 
                }
            }
            set
            {
                if (this.totalPhysicalQuantity_ != value)
                {
                    this.totalPhysicalQuantity_ = value;
                }
            }
        }
        #endregion
        
        public string totalPhysicalQuantityIDRef_ { get; set; }
        #region MinPhysicalQuantity_
        private List<CommodityNotionalQuantity> minPhysicalQuantity_;
        public List<CommodityNotionalQuantity> MinPhysicalQuantity_
        {
            get
            {
                if (this.minPhysicalQuantity_ != null)
                {
                    return this.minPhysicalQuantity_; 
                }
                else if (this.minPhysicalQuantityIDRef_ != null)
                {
                    return this.minPhysicalQuantity_; 
                }
                else
                {
                      return this.minPhysicalQuantity_; 
                }
            }
            set
            {
                if (this.minPhysicalQuantity_ != value)
                {
                    this.minPhysicalQuantity_ = value;
                }
            }
        }
        #endregion
        
        public string minPhysicalQuantityIDRef_ { get; set; }
        #region MaxPhysicalQuantity_
        private List<CommodityNotionalQuantity> maxPhysicalQuantity_;
        public List<CommodityNotionalQuantity> MaxPhysicalQuantity_
        {
            get
            {
                if (this.maxPhysicalQuantity_ != null)
                {
                    return this.maxPhysicalQuantity_; 
                }
                else if (this.maxPhysicalQuantityIDRef_ != null)
                {
                    return this.maxPhysicalQuantity_; 
                }
                else
                {
                      return this.maxPhysicalQuantity_; 
                }
            }
            set
            {
                if (this.maxPhysicalQuantity_ != value)
                {
                    this.maxPhysicalQuantity_ = value;
                }
            }
        }
        #endregion
        
        public string maxPhysicalQuantityIDRef_ { get; set; }
        #region ElectingParty_
        private PartyReference electingParty_;
        public PartyReference ElectingParty_
        {
            get
            {
                if (this.electingParty_ != null)
                {
                    return this.electingParty_; 
                }
                else if (this.electingPartyIDRef_ != null)
                {
                    electingParty_ = IDManager.getID(electingPartyIDRef_) as PartyReference;
                    return this.electingParty_; 
                }
                else
                {
                      return this.electingParty_; 
                }
            }
            set
            {
                if (this.electingParty_ != value)
                {
                    this.electingParty_ = value;
                }
            }
        }
        #endregion
        
        public string electingPartyIDRef_ { get; set; }
        public string choiceStr_0;
        public string choiceStr_1;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/GasPhysicalQuantity.cpp
// GasPhysicalQuantity.cpp 
#include "GasPhysicalQuantity.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

GasPhysicalQuantity::GasPhysicalQuantity(TiXmlNode* xmlNode)
: CommodityPhysicalQuantityBase(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //physicalQuantityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* physicalQuantityNode = xmlNode->FirstChildElement("physicalQuantity");

   if(physicalQuantityNode){physicalQuantityIsNull_ = false;}
   else{physicalQuantityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- physicalQuantityNode , address : " << physicalQuantityNode << std::endl;
   #endif
   if(physicalQuantityNode)
   {
      if(physicalQuantityNode->Attribute("href") || physicalQuantityNode->Attribute("id"))
      {
          if(physicalQuantityNode->Attribute("id"))
          {
             physicalQuantityIDRef_ = physicalQuantityNode->Attribute("id");
             physicalQuantity_ = boost::shared_ptr<CommodityNotionalQuantity>(new CommodityNotionalQuantity(physicalQuantityNode));
             physicalQuantity_->setID(physicalQuantityIDRef_);
             IDManager::instance().setID(physicalQuantityIDRef_,physicalQuantity_);
          }
          else if(physicalQuantityNode->Attribute("href")) { physicalQuantityIDRef_ = physicalQuantityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { physicalQuantity_ = boost::shared_ptr<CommodityNotionalQuantity>(new CommodityNotionalQuantity(physicalQuantityNode));}
   }

   //physicalQuantityScheduleNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* physicalQuantityScheduleNode = xmlNode->FirstChildElement("physicalQuantitySchedule");

   if(physicalQuantityScheduleNode){physicalQuantityScheduleIsNull_ = false;}
   else{physicalQuantityScheduleIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- physicalQuantityScheduleNode , address : " << physicalQuantityScheduleNode << std::endl;
   #endif
   if(physicalQuantityScheduleNode)
   {
      if(physicalQuantityScheduleNode->Attribute("href") || physicalQuantityScheduleNode->Attribute("id"))
      {
          if(physicalQuantityScheduleNode->Attribute("id"))
          {
             physicalQuantityScheduleIDRef_ = physicalQuantityScheduleNode->Attribute("id");
             physicalQuantitySchedule_ = boost::shared_ptr<CommodityPhysicalQuantitySchedule>(new CommodityPhysicalQuantitySchedule(physicalQuantityScheduleNode));
             physicalQuantitySchedule_->setID(physicalQuantityScheduleIDRef_);
             IDManager::instance().setID(physicalQuantityScheduleIDRef_,physicalQuantitySchedule_);
          }
          else if(physicalQuantityScheduleNode->Attribute("href")) { physicalQuantityScheduleIDRef_ = physicalQuantityScheduleNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { physicalQuantitySchedule_ = boost::shared_ptr<CommodityPhysicalQuantitySchedule>(new CommodityPhysicalQuantitySchedule(physicalQuantityScheduleNode));}
   }

   //totalPhysicalQuantityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* totalPhysicalQuantityNode = xmlNode->FirstChildElement("totalPhysicalQuantity");

   if(totalPhysicalQuantityNode){totalPhysicalQuantityIsNull_ = false;}
   else{totalPhysicalQuantityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- totalPhysicalQuantityNode , address : " << totalPhysicalQuantityNode << std::endl;
   #endif
   if(totalPhysicalQuantityNode)
   {
      if(totalPhysicalQuantityNode->Attribute("href") || totalPhysicalQuantityNode->Attribute("id"))
      {
          if(totalPhysicalQuantityNode->Attribute("id"))
          {
             totalPhysicalQuantityIDRef_ = totalPhysicalQuantityNode->Attribute("id");
             totalPhysicalQuantity_ = boost::shared_ptr<UnitQuantity>(new UnitQuantity(totalPhysicalQuantityNode));
             totalPhysicalQuantity_->setID(totalPhysicalQuantityIDRef_);
             IDManager::instance().setID(totalPhysicalQuantityIDRef_,totalPhysicalQuantity_);
          }
          else if(totalPhysicalQuantityNode->Attribute("href")) { totalPhysicalQuantityIDRef_ = totalPhysicalQuantityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { totalPhysicalQuantity_ = boost::shared_ptr<UnitQuantity>(new UnitQuantity(totalPhysicalQuantityNode));}
   }

   //minPhysicalQuantityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* minPhysicalQuantityNode = xmlNode->FirstChildElement("minPhysicalQuantity");

   if(minPhysicalQuantityNode){minPhysicalQuantityIsNull_ = false;}
   else{minPhysicalQuantityIsNull_ = true;}

   if(minPhysicalQuantityNode)
   {
      for(minPhysicalQuantityNode; minPhysicalQuantityNode; minPhysicalQuantityNode = minPhysicalQuantityNode->NextSiblingElement("minPhysicalQuantity")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- minPhysicalQuantityNode , address : " << minPhysicalQuantityNode << std::endl;
          #endif
          if(minPhysicalQuantityNode->Attribute("href") || minPhysicalQuantityNode->Attribute("id"))
          {
              if(minPhysicalQuantityNode->Attribute("id"))
              {
                  minPhysicalQuantityIDRef_ = minPhysicalQuantityNode->Attribute("id");
                  minPhysicalQuantity_.push_back(boost::shared_ptr<CommodityNotionalQuantity>(new CommodityNotionalQuantity(minPhysicalQuantityNode)));
                  minPhysicalQuantity_.back()->setID(minPhysicalQuantityIDRef_);
                  IDManager::instance().setID(minPhysicalQuantityIDRef_, minPhysicalQuantity_.back());
              }
              else if(minPhysicalQuantityNode->Attribute("href")) { minPhysicalQuantityIDRef_ = minPhysicalQuantityNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { minPhysicalQuantity_.push_back(boost::shared_ptr<CommodityNotionalQuantity>(new CommodityNotionalQuantity(minPhysicalQuantityNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- minPhysicalQuantityNode , address : " << minPhysicalQuantityNode << std::endl;
       #endif
   }

   //maxPhysicalQuantityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* maxPhysicalQuantityNode = xmlNode->FirstChildElement("maxPhysicalQuantity");

   if(maxPhysicalQuantityNode){maxPhysicalQuantityIsNull_ = false;}
   else{maxPhysicalQuantityIsNull_ = true;}

   if(maxPhysicalQuantityNode)
   {
      for(maxPhysicalQuantityNode; maxPhysicalQuantityNode; maxPhysicalQuantityNode = maxPhysicalQuantityNode->NextSiblingElement("maxPhysicalQuantity")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- maxPhysicalQuantityNode , address : " << maxPhysicalQuantityNode << std::endl;
          #endif
          if(maxPhysicalQuantityNode->Attribute("href") || maxPhysicalQuantityNode->Attribute("id"))
          {
              if(maxPhysicalQuantityNode->Attribute("id"))
              {
                  maxPhysicalQuantityIDRef_ = maxPhysicalQuantityNode->Attribute("id");
                  maxPhysicalQuantity_.push_back(boost::shared_ptr<CommodityNotionalQuantity>(new CommodityNotionalQuantity(maxPhysicalQuantityNode)));
                  maxPhysicalQuantity_.back()->setID(maxPhysicalQuantityIDRef_);
                  IDManager::instance().setID(maxPhysicalQuantityIDRef_, maxPhysicalQuantity_.back());
              }
              else if(maxPhysicalQuantityNode->Attribute("href")) { maxPhysicalQuantityIDRef_ = maxPhysicalQuantityNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { maxPhysicalQuantity_.push_back(boost::shared_ptr<CommodityNotionalQuantity>(new CommodityNotionalQuantity(maxPhysicalQuantityNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- maxPhysicalQuantityNode , address : " << maxPhysicalQuantityNode << std::endl;
       #endif
   }

   //electingPartyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* electingPartyNode = xmlNode->FirstChildElement("electingParty");

   if(electingPartyNode){electingPartyIsNull_ = false;}
   else{electingPartyIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- electingPartyNode , address : " << electingPartyNode << std::endl;
   #endif
   if(electingPartyNode)
   {
      if(electingPartyNode->Attribute("href") || electingPartyNode->Attribute("id"))
      {
          if(electingPartyNode->Attribute("id"))
          {
             electingPartyIDRef_ = electingPartyNode->Attribute("id");
             electingParty_ = boost::shared_ptr<PartyReference>(new PartyReference(electingPartyNode));
             electingParty_->setID(electingPartyIDRef_);
             IDManager::instance().setID(electingPartyIDRef_,electingParty_);
          }
          else if(electingPartyNode->Attribute("href")) { electingPartyIDRef_ = electingPartyNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { electingParty_ = boost::shared_ptr<PartyReference>(new PartyReference(electingPartyNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<CommodityNotionalQuantity> GasPhysicalQuantity::getPhysicalQuantity()
{
   if(!this->physicalQuantityIsNull_){
        if(physicalQuantityIDRef_ != ""){
             return boost::shared_static_cast<CommodityNotionalQuantity>(IDManager::instance().getID(physicalQuantityIDRef_));
        }else{
             return this->physicalQuantity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityNotionalQuantity>();
   }
}
boost::shared_ptr<CommodityPhysicalQuantitySchedule> GasPhysicalQuantity::getPhysicalQuantitySchedule()
{
   if(!this->physicalQuantityScheduleIsNull_){
        if(physicalQuantityScheduleIDRef_ != ""){
             return boost::shared_static_cast<CommodityPhysicalQuantitySchedule>(IDManager::instance().getID(physicalQuantityScheduleIDRef_));
        }else{
             return this->physicalQuantitySchedule_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityPhysicalQuantitySchedule>();
   }
}
boost::shared_ptr<UnitQuantity> GasPhysicalQuantity::getTotalPhysicalQuantity()
{
   if(!this->totalPhysicalQuantityIsNull_){
        if(totalPhysicalQuantityIDRef_ != ""){
             return boost::shared_static_cast<UnitQuantity>(IDManager::instance().getID(totalPhysicalQuantityIDRef_));
        }else{
             return this->totalPhysicalQuantity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<UnitQuantity>();
   }
}
std::vector<boost::shared_ptr<CommodityNotionalQuantity>> GasPhysicalQuantity::getMinPhysicalQuantity()
{
   if(!this->minPhysicalQuantityIsNull_){
        if(minPhysicalQuantityIDRef_ != ""){
             return this->minPhysicalQuantity_;
        }else{
             return this->minPhysicalQuantity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<CommodityNotionalQuantity>>();
   }
}
std::vector<boost::shared_ptr<CommodityNotionalQuantity>> GasPhysicalQuantity::getMaxPhysicalQuantity()
{
   if(!this->maxPhysicalQuantityIsNull_){
        if(maxPhysicalQuantityIDRef_ != ""){
             return this->maxPhysicalQuantity_;
        }else{
             return this->maxPhysicalQuantity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<CommodityNotionalQuantity>>();
   }
}
boost::shared_ptr<PartyReference> GasPhysicalQuantity::getElectingParty()
{
   if(!this->electingPartyIsNull_){
        if(electingPartyIDRef_ != ""){
             return boost::shared_static_cast<PartyReference>(IDManager::instance().getID(electingPartyIDRef_));
        }else{
             return this->electingParty_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PartyReference>();
   }
}
}

